caikit/core/data_model/base.py

```diff
@@ -344,10 +415,22 @@ class DataBase(metaclass=_DataBaseMetaClass):
     """
 
     def __setattr__(self, name, val):
-        """If attempting to set one of the named fields, instead set the
-        "private" version of the attribute.
+        """Handle attribute setting for oneofs and named fields with delegation
+        to backends as needed
         """
-        if name in self.__class__.fields:
+        # If setting a oneof directly, remove any oneof information
```

Comment:
This is needed so that the next time `which_oneof` is called, it will redo the lazy introspection by type